codice/ddf
# DDF-4356 Workspace Query Data Migration

#### What does this PR do?
- Data migration framework and karaf commands
- Data migration for workspace/query separation

#### Who is reviewing it? 

@leo-sakh 
@jrnorth 
@Corey-Collins 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@rzwiefel

#### How should this be tested?
- Build a version of DDF before this commit (336b27f9765d7b46bb4c6ca167a87c2d1d5d3ddc)
- Create a few workspaces and add varying amounts of queries to each
- Perform a catalog dump
- Build this branch and perform a catalog ingest
- Run migration commands
- Verify that data migration was successful

- Alternatively just ask me for some test data to ingest

#### Any background context you want to provide?
Previously queries were embedded in their workspaces. Queries are now their own entity and only linked to workspaces via unidirectional association. Given that this changes how the data is stored you may need to migrate all of the data in an existing system. These commands will perform that operation. 

#### What are the relevant tickets?
[DDF-4356](https://codice.atlassian.net/browse/DDF-4356)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
